packages/core/src/lib/lit-core.ts

```diff
@@ -301,7 +301,41 @@ export class LitCore {
       stakingContract.on('StateChanged', async (state: StakingStates) => {
         log(`New state detected: "${state}"`);
         if (state === StakingStates.NextValidatorSetLocked) {
+          log(
+            'State found to be new validator set locked, checking validator set'
+          );
+          let oldNodeUrls: string[] = [...this.config.bootstrapUrls].sort();
           await this.setNewConfig();
+          let currentNodeUrls: string[] = this.config.bootstrapUrls.sort();
+          let delta: string[] = currentNodeUrls.filter(
+            (item) => oldNodeUrls.indexOf(item) > -1
+          );
```

Comment:
Just some details:
- there is no need to sort the arrays
- prefer `const` over `let`
- instead of using `Array.indexOf` we could use a Set, Map/object or `Array.includes`

Reply:
Good call, yes quite right. I have been used to using `indexOf` for a long time. switch to `includes` but kept the top level `filter`.